Automattic/newspack-plugin
# Advertising Wizard: Google Ad Manager integration

### All Submissions:


### Changes proposed in this Pull Request:

Includes Google Ad Manager support in the Advertising wizard.

### How to test the changes in this Pull Request:

1. Apply the patch
2. Run `npm ci && npm run clean && npm run build:webpack`
3. Observe a new item - Google Ad Manager - in the Advertising Checklist
4. Enter the wizard and add, edit, update and delete ad slots

Note that the data model here uses post types. I explored two other options:

1. A single option containing all ad codes. This fell down because it meant manually maintaining IDs for each ad slot in order for them to be successfully referenced on the front end later on.
2. An option for each ad (with an ID in the option name). This sufferes from the same problem as above, but with the added complication that getting _all_ ad slots would need a pretty nasty `LIKE` query on the options table.

Once this is in a separate PR will introduce the widget/block for inserting an ad slot somewhere into the site.

### Other information:
